### PR TITLE
Diagram/setup file structure

### PR DESCRIPTION
**Pull request by jayeclark**
Partially fixes #8 

OK, I made great progress today and have a working prototype of a Diagram class along the lines of what you described. I've got three pull requests set up in sequential branches, that I think will help make it a little less overwhelming to review & approve the changes. (I'm going to close the other draft PR as this is the better way to go about it.)

1. diagram/setup-file-structure --- literally just copy/pasting files over to a new diagram folder where I'll be working and consolidating functions. I've done it this way because the goal is to have Diagram wholly separate from the website and exportable to the npm registry or as minified js.
2. diagram/create-diagram-service --- (I will raise this PR after the first is merged) more copy/pasting, creating a js file that only contains diagram-related functions
3. diagram/refactor-to-class --- (I will raise this PR after 2 is merged) this one contains the most relevant changes. (The first two PRs basically exist just to minimize the code changes that need to be reviewed in the 3rd PR and future PRs on the diagram class.) 

3 is still a work in progress. Currently the diagram only renders the basic data, and is missing some features (for example, it doesn't have showPercentage fully set up yet), but, it DOES work and renders a diagram to the diagramFrame div when `new Diagram()` is called with basic data sets passed as arguments. I figure I can add a few more commits fixing the last few todos in the file, and then pass it off to the other contributor who wanted to work on converting to Typescript. (Maybe he can do some refactoring as well - there is a lot of room for simplification!)

### CODE REVIEW

File: diagram/SetService.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export default class SetService{
```
Comment (heberleh) [inline diagram/SetService.js:1]:
```javascript
class SetService{}
```
then
```javascript
export default new SetService();
```
allows you to use the same object in the entire app, like this:
```javascript
import setService from 'SetService'; // lowercase setService (object)
```

**Review by heberleh — APPROVED** [review]
I made a review about the 'service' class

File: diagram/SetService.js (hunk 1 above)
Comment (jayeclark) [reply diagram/SetService.js:1]: I actually do take care of the export in the next PR (or maybe the third one? I can’t remember but it does turn into a default export at some point along the way) but I’m all for moving it to lowercase and trying to keep the convention that only classes are capitalized.

Comment (jayeclark) [reply diagram/SetService.js:1]: Ohhh, sorry, I was reading this on my phone earlier and it was hard to tell what the suggestion was. Yes, this change makes sense - there's no reason to have to create new instances of the object in each file it's imported into.

**Pull request merged**